python magic for jvm kernels

lmitusinski opened the pull request.

Commit:
#7124 python magic for jvm kernels

Commit:
Merge remote-tracking branch 'remotes/origin/master' into lmitusinski/7124

# Conflicts:
#	kernel/base/src/main/java/com/twosigma/beakerx/kernel/Kernel.java
#	kernel/base/src/main/java/com/twosigma/beakerx/kernel/KernelFunctionality.java
#	kernel/base/src/test/java/com/twosigma/beakerx/KernelTest.java

Commit:
#7124 python magic - fixed json and indents handling

Commit:
#7124 handling py4j python process within jvm kernels

Commit:
#7124 removed gradle task for moving py4j script

Commit:
#7124 handling communication with nested kernel in python magic

Commit:
#7124 automatic port selection for jvm-python communication

Commit:
Merge remote-tracking branch 'remotes/origin/master' into lmitusinski/7124

scottdraves commented on beakerx/beakerx_magics/python_magic.py:
```diff
@@ -0,0 +1,89 @@
+from py4j.clientserver import ClientServer, JavaParameters, PythonParameters
```
needs banner

scottdraves commented on kernel/base/src/main/java/com/twosigma/beakerx/kernel/PythonMagicManager.java:
```diff
@@ -0,0 +1,176 @@
+public class PythonMagicManager {
+
+    private static String PY4J_SCRIPT_NAME = "/beakerx_magics/python_magic.py";
+    private static String PYTHON = "python3";
```
"python" do not need to say 3

scottdraves commented on kernel/base/src/main/java/com/twosigma/beakerx/kernel/PythonMagicManager.java:
```diff
@@ -0,0 +1,176 @@
+        //cleanup communication resources if already in use
+        exit();
+
+        String home = System.getenv(ENV_BEAKERX_HOME);
```
what is this envar doing?

scottdraves commented on kernel/base/src/main/java/com/twosigma/beakerx/kernel/PythonMagicManager.java:
```diff
@@ -0,0 +1,176 @@
+                String pyScriptPath = new File(home).getParentFile().toString();
+                String[] cmd = {
+                        PYTHON,
+                        pyScriptPath + PY4J_SCRIPT_NAME,
```
why not just run a cmd like `beakerx start_server --port port --pyport port`

scottdraves commented on kernel/base/src/main/java/com/twosigma/beakerx/kernel/magic/command/functionality/PythonMagicCommand.java:
```diff
@@ -0,0 +1,79 @@
+        List<Message> messages = new ArrayList<>();
+        while (true) {
+            String iopubMsg = pep.getIopubMsg();
+            if (iopubMsg.equals("null")) break;
```
what does this do?

scottdraves commented on kernel/base/src/test/java/com/twosigma/beakerx/KernelTest.java:
```diff
@@ -31,6 +31,8 @@
 import com.twosigma.beakerx.kernel.KernelManager;
 import com.twosigma.beakerx.kernel.PathToJar;
+import com.twosigma.beakerx.kernel.PythonEntryPoint;
+import com.twosigma.beakerx.kernel.PythonMagicManager;
```
imports added but no test?

Commit:
#7124 code review fixes
-added missing banner
-additonal comments
-beakerx py4jserver
-waiting for process initialization
-python magic test

scottdraves committed:
remove dead code
